include/ensmallen_bits/callbacks/callbacks.hpp

```diff
@@ -744,6 +744,101 @@ class Callback
                     MatType& /* coordinates */)
   { return false; }
 
+ /**
+  * Invoke the StepTaken() callback if it exists.
+  * Specialization for MultiObjective case.
+  *
+  * @param callback The callback to call.
+  * @param optimizer The optimizer used to update the function.
+  * @param function Function to optimize.
+  * @param coordinates Starting point.
+  * @param calculatedObjectives The set of calculated objectives so far.
+  * @param frontIndices The indices of the members belonging to Pareto Front.
+  */
+  template<typename CallbackType,
+           typename OptimizerType,
+           typename FunctionType,
+           typename MatType,
+           typename ObjectivesArrayType,
+           typename IndicesType>
+    static typename std::enable_if<
+    callbacks::traits::HasMOOStepTakenSignature<
+    CallbackType, OptimizerType, FunctionType, MatType, ObjectivesArrayType,
+    IndicesType>::hasBool, bool>::type
+    StepTakenFunction(CallbackType& callback,
+      OptimizerType& optimizer,
+      FunctionType& function,
+      MatType& coordinates,
+      ObjectivesArrayType& calculatedObjectives,
```

Comment:
I would call this `ObjectivesVecType` and `objectives`, everything we pass is calculated in some form so no need to add a prefix for that at the same time this is closer to the existing `StepTakenFunction`.

Reply:
It is indeed, but i wanted to differentiate between then because it will cause very long codes and might be polluting. 

So, I created a separate MOOTypedForms specifically for MOO based works and then did ```hasBool``` and ```hasVoid``` in them. 

If I were to add this in TypedForms I would have to add

```hasMOObool``` and ```hasMOOVoid```. In my way, we get to not touch our old implementation at all, and add these functionalities.

Reply:
Not sure I see the connection, I was referring to the naming of the parameters here.

Reply:
I was clarifying as to why its similar to ```StepTakenFunction```, I've applied your suggestion on the name change already :+1:. Looks like an instance of miscommunication, haha